includes/core-changes/aspnetcore/3.1/http-cookie-samesite-authn-impacts.md

```diff
@@ -0,0 +1,157 @@
+
+##### Microsoft Edge
+
+While Microsoft Edge supports the old `SameSite` standard, as of version 44 it didn't have any compatibility problems with the new standard.
+
+##### Microsoft Edge Chromium
+
+The feature flag is `edge://flags/#same-site-by-default-cookies`. No compatibility issues were observed when testing with Microsoft Edge Chromium 78.
+
+##### Electron
+
+Versions of Electron include older versions of Chromium. For example, the version of Electron used by Microsoft Teams is Chromium 66, which exhibits the older behavior. Perform your own compatibility testing with the version of Electron your product uses. For more information, see [Support older browsers](#support-older-browsers).
+
+##### Support older browsers
+
+The 2016 `SameSite` standard mandated that unknown values must be treated as `SameSite=Strict` values. Consequently, any older browsers that support the original standard may break when they see a `SameSite` property with a value of `None`. Web apps must implement browser sniffing if they intend to support these old browsers. ASP.NET Core doesn't implement browser sniffing for you because `User-Agent` request header values are highly unstable and change on a weekly basis. What is available is an extension point in `CookiePolicy` allowing you to plug in `User-Agent`-specific logic.
+
+In *Startup.cs*, add the following code:
+
+```csharp
+private void CheckSameSite(HttpContext httpContext, CookieOptions options) 
+{
+    if (options.SameSite == SameSiteMode.None) 
+    { 
+        var userAgent = httpContext.Request.Headers["User-Agent"].ToString(); 
+        // TODO: Use your User Agent library of choice here. 
+        if (/* UserAgent doesn't support new behavior */) 
+        { 
+            options.SameSite = SameSiteMode.Unspecified; 
+        }
+    }
+}
+
+public void ConfigureServices(IServiceCollection services) 
+{ 
+    services.Configure<CookiePolicyOptions>(options => 
+    { 
+        options.MinimumSameSitePolicy = SameSiteMode.Unspecified; 
+        options.OnAppendCookie = cookieContext =>
+            CheckSameSite(cookieContext.Context, cookieContext.CookieOptions); 
+        options.OnDeleteCookie = cookieContext =>
+            CheckSameSite(cookieContext.Context, cookieContext.CookieOptions);
+    }); 
+} 
+
+public void Configure(IApplicationBuilder app) 
+{ 
+    // Before UseAuthentication or anything else that writes cookies. 
+    app.UseCookiePolicy();
+
+    app.UseAuthentication(); 
+    // code omitted for brevity
+}
+```
+
+##### Opt-out switches
+
+The `Microsoft.AspNetCore.SuppressSameSiteNone` compatibility switch enables you to temporarily opt out of the new ASP.NET Core cookie behavior. Add the following JSON to a *runtimeconfig.template.json* file in your project:
+
+```json
+{ 
+  "configProperties": { 
+    "Microsoft.AspNetCore.SuppressSameSiteNone": "true" 
+  } 
+}
+```
+
+The preceding switch will be removed in the next major version of ASP.NET Core.
```

Comment:
We're not supposed to discuss future plans in docs - see the "Future product plans" bullet here: https://review.docs.microsoft.com/en-us/help/contribute/contribute-get-started-channel-guidance?branch=master#content-that-does-not-belong-in-a-docsmicrosoftcom-article-or-repository